saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -284,14 +278,20 @@ def perform_mutation(cls, _root, info, **data):
         return FulfillmentUpdateTracking(fulfillment=fulfillment, order=order)
 
 
+class FulfillmentCancelInput(graphene.InputObjectType):
+    warehouse_id = graphene.ID(
+        description="ID of warehouse where items will be restock.", required=False
```

Comment:
```suggestion
        description="ID of a warehouse where items will be restocked.", required=False
```

Reply:
We could extend this description and describe the validation logic for this argument - this argument is optional when given fulfillment has `WAITING_FOR_APPROVAL` status; it is required otherwise.